src/AutoRest.CSharp/Common/Input/CadlInputEnumTypeConverter.cs

```diff
@@ -103,7 +121,17 @@ private static IReadOnlyList<InputEnumTypeValue> normalizeValues(IReadOnlyList<I
                 case InputTypeKind.Float32:
                     foreach (var value in allowedValues)
                     {
-                        concreteValues.Add(new InputEnumTypeFloatValue(value.Name, (float)value.Value, value.Description));
+                        switch (value.Value)
+                        {
+                            case Int32 i:
+                                concreteValues.Add(new InputEnumTypeFloatValue(value.Name, (float)(Int32)i, value.Description));
+                                break;
+                            case Single single:
+                                concreteValues.Add(new InputEnumTypeFloatValue(value.Name, (float)single, value.Description));
```

Comment:
We do not need the cast because `single` here is already a `float`
```suggestion
                                concreteValues.Add(new InputEnumTypeFloatValue(value.Name, single, value.Description));
```